checker/src/abstract_value.rs

```diff
@@ -1197,7 +1209,7 @@ impl AbstractValueTrait for Rc<AbstractValue> {
             Expression::TaggedExpression { operand, tag } => {
                 operand.try_to_retype_as(target_type).add_tag(*tag)
             }
-            Expression::Variable { path, .. } => {
+            Expression::Variable { path, .. } | Expression::RefinedVariable { path, .. } => {
                 AbstractValue::make_typed_unknown(target_type.clone(), path.clone())
```

Comment:
For RefinedVariable, should this be a call to make_typed_refined_unknown?

Reply:
Ah yes, good catch.